### PR TITLE
Improve scripts (with npm and discord rpc)

### PR DESCRIPTION
**Pull request by Stylix58**
We have added NPM and Discord RPC to the fixed scripts.
I keep update scripts for pnpm because i use it on https://github.com/fosscord/fosscord-landingpage.

### CODE REVIEW

File: scripts/setup/setup.bat [hunk 1]
```diff
@@ -27,19 +27,19 @@ IF ERRORLEVEL 1 (
...
 IF ERRORLEVEL 1 (
 	ECHO Error: node is not installed.
-	ECHO Please Install NodeJS from: https://nodejs.org/en/download/
+	ECHO Please Install NodeJS from: https://nodejs.org/en/download
 	ECHO And make sure its in the path
     GOTO :end
 )
 
-where /q npm
+where /q pnpm
```
Comment (Flam3rboy) [inline scripts/setup/setup.bat:35]: you are only checking if pnpm is installed, please allow both npm OR pnpm

Comment (Stylix58) [reply scripts/setup/setup.bat]: I have fixed it.

File: scripts/update/update-npm.bat [hunk 2]
```diff
@@ -0,0 +1,3 @@
+@ECHO off
+ECHO Update for PNPM...
```
Comment (Flam3rboy) [inline scripts/update/update-npm.bat:2]: please change this to:
update all dependencies

File: scripts/setup/setup.sh [hunk 3]
```diff
@@ -61,7 +66,4 @@ git clone https://github.com/fosscord/react-native-withcss react-native-withcss
 
 echo '{"folders":[{"path":"overview"},{"path":"cdn"},{"path":"api"},{"path":"gateway"},{"path":"media"},{"path":"server-util"},{"path":"ui"},{"path":"client"},{"path":"plugins"},{"path":"themes"},{"path":"landingpage"},{"path":"dashboard"},{"path":"support"},{"path":"css-mediaquery"},{"path":"react-native-withcss"}]}' >> fosscord.code-workspace
 
-if [ -x "$(command -v code)" ]; then
```
Comment (Flam3rboy) [inline scripts/setup/setup.sh:64]: please add this again

File: scripts/setup/setup.sh [hunk 4]
```diff
@@ -25,17 +25,22 @@ if ! [ -x "$(command -v git)" ]; then
...
 if ! [ -x "$(command -v npm)" ]; then
   echo 'Error: npm is not installed.' >&2
-  echo Please install npm from: https://nodejs.org/en/download/
+  echo Please install npm from: https://nodejs.org/en/download
   echo And make sure its in the path
   exit 1
 fi
-echo ✓ Dependencies are already Installed
+if ! [ -x "$(command -v pnpm)" ]; then
+  echo PNPM is not installed but he is optionnal
```
Comment (Flam3rboy) [inline scripts/setup/setup.sh:39]: do not mention that the user can optionally install pnpm, just use pnpm if the user has it installed

File: scripts/setup/setup.bat [hunk 5]
```diff
@@ -27,19 +27,26 @@ IF ERRORLEVEL 1 (
...
-	ECHO Please install npm from: https://nodejs.org/en/download/
+	ECHO Please install npm from: https://nodejs.org/en/download
 	ECHO And make sure its in the path
     GOTO :end
 )
-echo Dependencies are already Installed
+
+where /q pnpm
+IF ERRORLEVEL 1 (
+	ECHO PNPM is not installed but he is optionnal
```
Comment (Flam3rboy) [inline scripts/setup/setup.bat:45]: same here, do not mention that pnpm can be installed optionally, instead just use it if its available

**Review by Flam3rboy — CHANGES_REQUESTED** [review]

File: scripts/setup/setup.bat (hunk 5 above)
Comment (Stylix58) [reply scripts/setup/setup.bat:45]: I think this should be hard to code this but i will try (any help would be appreciated)

Comment (xNaCly) [reply scripts/setup/setup.bat:45]: also change the wording from he to it

**Comment by Stylix58** [comment]
I have made the requested changes!

File: README.md [hunk 6]
```diff
@@ -27,7 +27,13 @@ logo by [@nwlandas](https://twitter.com/nwlandas)
 
 ## Installation
 
-_it is in development and not yet finished_
```
Comment (Flam3rboy) [inline README.md:30]: why did you remove this?
This is for non tech people who want to try fosscord and there is not a beta version yet.

File: README.md [hunk 7]
```diff
@@ -27,7 +27,13 @@ logo by [@nwlandas](https://twitter.com/nwlandas)
...
+### Windows
```
Comment (Flam3rboy) [inline README.md:30]: Please move the install instructions to the scripts/setup directory
they are for contrubitors

File: scripts/setup/setup.bat [hunk 8]
```diff
@@ -27,19 +27,26 @@ IF ERRORLEVEL 1 (
...
-	ECHO Please install npm from: https://nodejs.org/en/download/
+	ECHO Please install npm from: https://nodejs.org/en/download
 	ECHO And make sure its in the path
     GOTO :end
 )
-echo Dependencies are already Installed
+
+where /q pnpm
+IF ERRORLEVEL 1 (
+	ECHO PNPM is not installed but it is optionnal
```
Comment (Flam3rboy) [inline scripts/setup/setup.bat:45]: remove those mentions

**Review by Flam3rboy — CHANGES_REQUESTED** [review]

**Review by Stylix58 — COMMENTED** [review]
Fixed!

File: scripts/setup/README.md [hunk 9]
```diff
@@ -1,17 +1,9 @@
...
+# Getting started
 
 ### Windows
+1. Open up a Terminal.
+2. Type this command: `git clone https://github.com/fosscord/fosscord && .\scripts\setup\setup.bat`.
 
-to setup all repositories open cmd and enter this to command
-
-```
-curl https://raw.githubusercontent.com/fosscord/fosscord/main/scripts/setup/setup.bat --output setup.bat && setup.bat
```
Comment (Flam3rboy) [inline scripts/setup/README.md:8]: I think curl is better, because It'll only download the setup file and setup alle repositories in the root folder not in a sub dir of the overview repo

Comment (Flam3rboy) [reply scripts/setup/README.md:8]: Also curl is installed per default in windows 10

**Review by Flam3rboy — CHANGES_REQUESTED** [review]

**Review by Stylix58 — COMMENTED** [review]
Fixed (and improved also!)

File: scripts/setup/README.md [hunk 10]
```diff
@@ -1,17 +1,14 @@
-## Getting Started
+# Getting started
+## Wihout automated scripts (for all systems)
```
Comment (Flam3rboy) [inline scripts/setup/README.md:2]: typo xD

**Comment by Stylix58** [comment]
resolved

File: scripts/setup/README.md [hunk 11]
```diff
@@ -1,4 +1,4 @@
-# Getting started
+# Getting Started
 ## Wihout automated scripts (for all systems)
```
Comment (Flam3rboy) [inline scripts/setup/README.md:2]: typo

**Review by Flam3rboy — CHANGES_REQUESTED** [review]

**Review by Stylix58 — COMMENTED** [review]
RESOLVED

File: scripts/setup/README.md [hunk 12]
```diff
@@ -1,17 +1,14 @@
-## Getting Started
+# Getting Started
+## Without automated scripts (for all systems)
+1. Open up a Terminal.
+1. Type this command: `curl https://raw.githubusercontent.com/fosscord/fosscord/main/scripts/setup/setup.bat --output setup.bat && setup.bat`.
```
Comment (Flam3rboy) [inline scripts/setup/README.md:4]: do .bat files work on mac/linux?

File: scripts/update/update-git.sh [hunk 13]
```diff
@@ -1,11 +1,9 @@
-echo Update all repositories ...
-cd ../../../
+echo Update files...
+cd ../..
 for D in */; do
 	echo --------------
 	echo "$D";
 	cd $D
-	git pull
```
Comment (Flam3rboy) [inline scripts/update/update-git.sh:7]: I think this is wrong, the ``update-git.sh`` should pull all new changes?

File: scripts/setup/README.md (hunk 12 above)
Comment (Stylix58) [reply scripts/setup/README.md]: FIXED (xd)

File: scripts/update/update-git.sh (hunk 13 above)
Comment (Stylix58) [reply scripts/update/update-git.sh:7]: `git sync`?

Comment (Stylix58) [reply scripts/update/update-git.sh:7]: I will use it

**Review by Stylix58 — COMMENTED** [review]
Fixed
